vyper/codegen/ir_node.py

```diff
@@ -330,6 +330,15 @@ def is_complex_ir(self):
             and self.value.lower() not in do_not_cache
         )
 
+    # set an error message and push down into all children.
+    # useful for overriding an error message generated by a helper
+    # function with a more specific error message.
+    def set_error_msg(self, error_msg: str) -> None:
+        self.error_msg = error_msg
+        for arg in self.args:
+            self.set_error_msg(error_msg)
```

Comment:
What's going on here?

Reply:
it pushes down the error message into all the child IR nodes